stack.yaml

```diff
@@ -26,9 +26,12 @@ extra-deps:
 - hie-bios-0.3.2
 - hlint-2.2.8
 - hsimport-0.11.0
+- HsYAML-0.2.1.0
+- HsYAML-aeson-0.2.0.0
```

Comment:
You'll likely also need to add these to each `stack.yaml` and to `haskell-ide-engine.cabal`

Reply:
This doesn't seem to be working :-(

Reply:
I have to note that HsYAML is using GPL and the project licensing would be affected (not sure if it already is using libs with gpl)

Reply:
Personally i am fine with linking a gpl lib but the project will have two yaml libs and maybe we should use only one (not in this pr)

Reply:
I'm also okay with GPL, however stylish-haskell just made the switch, we could ask them to switch back https://github.com/jaspervdj/stylish-haskell/commit/498d676c1e06af966c814e01967fc3ece1fd9225

Reply:
I am not sure "fine" is the right wording, since we are not GPL, we do not really have the rights to using it, right? On the other hand, I doubt, too, that someone will sue us.

Reply:
Hey, I wasn't aware that HsYAML was GPL-licensed.  I don't think it's a problem -- it's fine for a BSD3 application to depend on a GPL-licensed library.  It does not affect the code of `haskell-ide-engine` or `stylish-haskell`, those remain BSD3-licensed.  However, the compiled binary must adhere to the GPL license (which seems fine in this case).  This is similar to e.g. the fact that GHC links against `libgmp` by default.

But, if this worries you, I am happy to add a flag to `stylish-haskell` that makes it pick the C-based `yaml` as a dependency rather than `HsYAML`?

Reply:
Afaiu we can use it without changing our licenses files but the license for hie artifacts linking HsYaml becomes auto gpl. So anybody could issue to someone who doesnt respect the gpl resdistributing hie.

Reply:
IANAL, if you argue that this is indeed just fine, I have nothing to say against it :)

Reply:
Ah. Assuming this is correct, this sounds totally reasonable.